### PR TITLE
Kevinf security

### PR DESCRIPTION
**Pull request by kfansler**
Security rewrite requested by Devin. I'll add him as a reviewer once I know his GitHub account name.

### CODE REVIEW

File: source/_introduction.html.md.erb [hunk 1]
```diff
@@ -263,9 +270,50 @@ The access token must be sent with every API call in an HTTP authorization heade
...
+The following provides some best practices to consider when working with the Smartsheet API and any API keys, tokens, or other sensitive information.
+
+#### Source Code and Version control
+
+Be wary of committing API keys to accessible version control systems like GitHub or BitBucket.
 
-<aside class="notice">If your application requires users to be able to login with their own account, you must implement the full OAuth flow. See the documentation for <a href="#third-party-app-development">Third-party App Development</a> and <a href="https://developers.smartsheet.com/register">Developer Registration</a>.</aside>
+* Use server environment variables. 
+* Use a config file excluded from source control, such as adding that filename to the .gitignore file.
+* If making these changes to existing code, be sure to regenerate the API keys so the previous versions can't be found in your code base.
```
Comment (drychetnik) [inline source/_introduction.html.md.erb]: Not sure I follow this one as written. Did you mean?

* If making these changes to existing code, be sure to regenerate the API keys and revoke the previous ones, which can still be seen in your version control system's history

**Review by drychetnik — COMMENTED** [review]
Mostly looks good to me. Just have the one comment where I didn't follow what was written.

**Review by timwellswa — COMMENTED** [review]
Kevin, it might be stated elsewhere but if it isn't as a best practice during OAuth we should tell developers to review the required access for their app and only request the necessary access scope. I have seen lots of examples where people just throw in all scopes if they aren't sure.

File: source/_introduction.html.md.erb [hunk 2]
```diff
@@ -263,9 +270,50 @@ The access token must be sent with every API call in an HTTP authorization heade
...
+### Security Best Practices
+
+The following provides some best practices to consider when working with the Smartsheet API and any API keys, tokens, or other sensitive information.
+
+#### Source Code and Version control
+
+Be wary of committing API keys to accessible version control systems like GitHub or BitBucket.
 
-<aside class="notice">If your application requires users to be able to login with their own account, you must implement the full OAuth flow. See the documentation for <a href="#third-party-app-development">Third-party App Development</a> and <a href="https://developers.smartsheet.com/register">Developer Registration</a>.</aside>
+* Use server environment variables.
```
Comment (timwellswa) [inline source/_introduction.html.md.erb]: Would drop server from this to just, `use environment variables`. Would put a note in here that if using one of the SDKs, the environment variable SMARTSHEET_ACCESS_TOKEN can be used in place of passing in an API token to the Smartsheet client initialization/client builder code. If the access token is null on input to the client builder, the SDK will automatically pick up the value of the SMARTSHEET_ACCESS_TOKEN environment variable.

**Comment by timwellswa** [comment]
The SDKs should really be handling the token refresh for the apps. We aren't there yet, so there is no immediate action item for this update, but we should discuss that as an SDK backlog item.

File: source/_introduction.html.md.erb (hunk 1 above)
Comment (kfansler) [reply source/_introduction.html.md.erb]: Yes. Your rewrite makes more sense. I'll fix. Thanks!

**Review by drychetnik — APPROVED** [review]

**Review by timwellswa — APPROVED** [review]
Looks good to me

**Pull request merged**